server/src/main/java/org/tctalent/server/service/db/util/PdfHelper.java

```diff
@@ -71,34 +77,69 @@ public Resource generatePdf(Candidate candidate, Boolean showName, Boolean showC
             String xHtml = convertToXhtml(renderedHtmlContent);
 
             // Remove any null bytes to avoid an invalid XML character (Unicode: 0x0) error
-            xHtml = Pattern.compile("\\x00").matcher(xHtml).replaceAll("");
-
-            ITextRenderer renderer = new ITextRenderer();
-
-            renderer.setDocumentFromString(xHtml, "classpath:pdf/");
-            renderer.layout();
+            xHtml = NULL_BYTE_PATTERN.matcher(xHtml).replaceAll("");
 
             // And finally, we create the PDF:
-            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
-            renderer.createPDF(outputStream);
-            outputStream.close();
-            return new ByteArrayResource(outputStream.toByteArray());
+            return createPdf(xHtml);
+
+        } catch (Exception e) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While re-throwing the caught exception with modifications , information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)

Suggested remediation:
Consider passing the exception object either as  &#34;rootCause&#34; or as an inner exception parameter, to the constructor of the new exception before throwing new exception. This will help keep stack trace.
```
@@ -90,3 +90,3 @@
 
-           throw new PdfGenerationException(e.getMessage());
+           throw new PdfGenerationException(e.getMessage(), e);
         }

```